components/script/dom/promise.rs

```diff
@@ -411,10 +411,14 @@ impl FromJSValConvertibleRc for Promise {
             return Ok(ConversionResult::Failure("not an object".into()));
         }
         rooted!(in(cx) let obj = value.get().to_object());
-        if !IsPromiseObject(obj.handle()) {
-            return Ok(ConversionResult::Failure("not a promise".into()));
-        }
-        let promise = Promise::new_with_js_promise(obj.handle(), SafeJSContext::from_ptr(cx));
+
+        let promise_global = GlobalScope::from_object_maybe_wrapped(obj.handle().get(), cx);
```

Comment:
Let's use GlobalScope::from_cx instead. That means we'll create the prise in the realm that is already active, rather than maybe changing to a different realm.